### DISCUSSION TITLE
@ExceptionHandler not getting called

### DISCUSSION THREAD

**Discussion by AnandMandhre** [graphql-java-kickstart/graphql-spring-boot#554] [Q&A]
Hello Guys, 

I have a problem trying to configure @ExceptionHandler with my springboot application. In my application exceptions get handled inside SimpleDataFetcherExceptionHandler despite configuring and @Component with @ExceptionHandler  methods for various exceptions.

I have set the properties correctly and followed the disucssion on link mentioned below.
**graphql.servlet.exception-handlers-enabled=true**

[https://github.com/graphql-java-kickstart/graphql-spring-boot/issues/384]

It could be that other jar file in my pom are causing it. I have included spring cloud stream and kafka binder recently. 

How can check if my exception handler got configured correctly (maybe via GraphQLWebAutoConfiguration or GraphQLErrorStartupListener)?

Is there a way to set it manually using @Configuration approach ?

Thanks

**Comment by AnandMandhre** [comment]
Debugging further and after setting debuging on. I noticed following logs

**DEBUG g.k.s.e.GraphQLErrorHandlerFactory - Handle GraphQL errors using exception handlers defined in 0 custom factories**

I had expected size of custom factories to be **1**. Having an object of type **ReflectiveGraphQLErrorFactory** containing my handler (bean). 

I came with idea to create GraphQLErrorFromExceptionHandler with List<GraphQLErrorFactory>, list could have contained my customerrorhandler but class is not visible out of the library.... so thats that. Running out of ideas :( 

Any suggestions group ?

**Reply by shemama2** [reply]
Have you resolved this issue?

**Reply by AnandMandhre** [reply]
No, I had to work around it. Using

@Component
public class CustomErrorHandler implements GraphQLErrorHandler 
@Override
	public List<GraphQLError> processErrors(List<GraphQLError> errors) {
          creating and returning GenericGraphQLError object with my custom message in it.

**Comment by oliemansm** [comment]
Are you using the latest version of this library? There was a problem with the exception handler for a while that went unnoticed in earlier versions. 

Did you add the `value` parameter to the `@ExceptionHandler` annotation? The implementation used here needs that value instead of being able to determine the type to use based on the parameter.

Does your annotated method have a return type of `GraphQLError` or a `List<GraphQLError>`? That's required to be taken into consideration.
